ASM/c/debug.c

```diff
@@ -31,71 +38,217 @@ warp_t dungeon_warps[] = {
 };
 
 item_t items_debug[] = {
     {  0, Z64_ITEM_HOOKSHOT, "Hookshot"},
     {  1, Z64_ITEM_LONGSHOT, "Longshot"},
-    {  2, Z64_ITEM_BOW, "Bow"},
-    {  3, Z64_ITEM_BOMB_BAG_20, "Bomb bag"},
-    {  4, Z64_ITEM_DEKU_SHIELD, "Deku Shield"},
-    {  5, Z64_ITEM_HYLIAN_SHIELD, "Hylian Shield"},
-    {  6, Z64_ITEM_MIRROR_SHIELD, "Mirror Shield"},
-    {  7, Z64_ITEM_FAIRY_OCARINA, "Ocarina"},
-    {  8, Z64_ITEM_BOMBCHU, "Bombchu"},
-    {  9, Z64_ITEM_BOOMERANG, "Boomerang"},
-    {  10, Z64_ITEM_HAMMER, "Hammer"},
-    {  11, Z64_ITEM_BOTTLE, "Bottle"},
-    {  12, Z64_ITEM_KOKIRI_SWORD, "KS"},
-    {  13, Z64_ITEM_BIGGORON_SWORD, "BGS"},
-    {  14, Z64_ITEM_IRON_BOOTS, "Irons"},
-    {  15, Z64_ITEM_HOVER_BOOTS, "Hovers"},
-    {  16, Z64_ITEM_GORONS_BRACELET, "Str1"},
-    {  17, Z64_ITEM_SILVER_GAUNTLETS, "Str2"},
-    {  18, Z64_ITEM_GOLDEN_GAUNTLETS, "Str3"},
-    {  19, Z64_ITEM_SILVER_SCALE, "Scale"},
-    {  20, Z64_ITEM_FOREST_MEDALLION, "Forest Med"},
-    {  21, Z64_ITEM_FIRE_MEDALLION, "Fire Med"},
-    {  22, Z64_ITEM_WATER_MEDALLION, "Water Med"},
-    {  23, Z64_ITEM_SPIRIT_MEDALLION, "Spirit Med"},
-    {  24, Z64_ITEM_SHADOW_MEDALLION, "Shadow Med"},
-    {  25, Z64_ITEM_LIGHT_MEDALLION, "Light Med"},
-    {  26, Z64_ITEM_KOKIRIS_EMERALD, "Emerald"},
-    {  27, Z64_ITEM_GORONS_RUBY, "Ruby"},
-    {  28, Z64_ITEM_ZORAS_SAPPHIRE, "Sapphire"},
+    {  2, Z64_ITEM_SLINGSHOT, "Slingshot"},
+    {  3, Z64_ITEM_BOW, "Bow"},
+    {  4, Z64_ITEM_BOMB_BAG_20, "Bomb bag"},
+    {  5, Z64_ITEM_DEKU_SHIELD, "Deku Shield"},
+    {  6, Z64_ITEM_HYLIAN_SHIELD, "Hylian Shield"},
+    {  7, Z64_ITEM_MIRROR_SHIELD, "Mirror Shield"},
+    {  8, Z64_ITEM_FAIRY_OCARINA, "Ocarina"},
+    {  9, Z64_ITEM_BOMBCHU, "Bombchu"},
+    {  10, Z64_ITEM_BOOMERANG, "Boomerang"},
+    {  11, Z64_ITEM_HAMMER, "Hammer"},
+    {  12, Z64_ITEM_BOTTLE, "Bottle"},
+    {  13, Z64_ITEM_KOKIRI_SWORD, "Kokiri Sword"},
+    {  14, Z64_ITEM_BIGGORON_SWORD, "Biggoron Sword"},
+    {  15, Z64_ITEM_IRON_BOOTS, "Iron boots"},
+    {  16, Z64_ITEM_HOVER_BOOTS, "Hover boots"},
+    {  17, Z64_ITEM_GORONS_BRACELET, "Strength 1"},
+    {  18, Z64_ITEM_SILVER_GAUNTLETS, "Strength 2"},
+    {  19, Z64_ITEM_GOLDEN_GAUNTLETS, "Strength 3"},
+    {  20, Z64_ITEM_SILVER_SCALE, "Scale"},
+    {  21, Z64_ITEM_NULL, "Magic"},
+    {  22, Z64_ITEM_FOREST_MEDALLION, "Forest Med"},
+    {  23, Z64_ITEM_FIRE_MEDALLION, "Fire Med"},
+    {  24, Z64_ITEM_WATER_MEDALLION, "Water Med"},
+    {  25, Z64_ITEM_SPIRIT_MEDALLION, "Spirit Med"},
+    {  26, Z64_ITEM_SHADOW_MEDALLION, "Shadow Med"},
+    {  27, Z64_ITEM_LIGHT_MEDALLION, "Light Med"},
+    {  28, Z64_ITEM_KOKIRIS_EMERALD, "Emerald"},
+    {  29, Z64_ITEM_GORONS_RUBY, "Ruby"},
+    {  30, Z64_ITEM_ZORAS_SAPPHIRE, "Sapphire"},
+};
+
+menu_category_t actor_categories[] = {
+    {  0, "SWITCH"},
+    {  1, "BG"},
+    {  2, "PLAYER"},
+    {  3, "EXPLOSIVES"},
+    {  4, "NPC"},
+    {  5, "ENEMY"},
+    {  6, "PROP"},
```

Comment:
(Apparently this comment didn't get saved in my first review)

![image](https://github.com/user-attachments/assets/0ce477c3-691d-4191-a235-73ca48a00df6)
Selecting this action caused the game to crash. I had just warped to Hyrule Field from Lake as adult before selecting it. I tried again in a different location and it didn't crash, but tried once again with the same setup and it crashed again. This time it did a hard-crash, i.e. PJ64 was unable to catch it and recover, but also looks like it tried to display something. Maybe just too many actors in the list?

![image](https://github.com/user-attachments/assets/b67f1820-0cef-439d-9314-95d1e4798ea2)

Reply:
This was a too small text buffer issue as you expected. Since the addition of the X Y Z values for Rob, it overflowed the initial size i put.

Reply:
![image](https://github.com/user-attachments/assets/fe583a62-845c-41eb-8801-2d8e26d6b3a9)
Still crashing for me (I'm sure I have the latest version because everything else you said is changed)

Reply:
I increased it again from 0xA00 to 0x1000 and it's working now. If increasing this more won't have any adverse effects, you can just make that change and I'll approve the PR (unfortunately I won't be able to test it again for the next week)
![image](https://github.com/user-attachments/assets/e686d507-420b-404f-a6f1-9c5c2ebdc278)